components/tools/OmeroWeb/omeroweb/webgateway/marshal.py

```diff
@@ -158,6 +158,17 @@ def imageMarshal(image, key=None, request=None):
     interpolate = server_settings.get('interpolate_pixels', True)
 
     try:
+        def pixel_size_in_microns(method):
+            try:
+                size = method('MICROMETER')
+                return size.getValue() if size else None
+            except:
+                logger.warn(
```

Comment:
Also as a potential follow-up of this PR, we could catch such conversion errors directly at the Python gateway level and let `ImageWrapper.getPixelSixe{XYZ}` return a dedicated exception e.g. `UnitConversionException` that could be consumed by consumers like OMERO.web gateway. Thoughts @aleksandra-tarkowska @will-moore ?